www/src/public/js/controllers/builder/main.js

```diff
@@ -1425,7 +1441,21 @@
         /** Confirm function for the edit character modal. */
         var editCharacter = function() {
             if (validateEditCharacter()) {
-                $scope.allLists[$scope.selectedListIndex].name = $scope.textInputModalModel.input;
+                var cookieConsent = $cookies.get("cookie-consent");
+                var oldName = $scope.allLists[$scope.selectedListIndex].name
+                var newName = $scope.textInputModalModel.input;
+
+                //copy savedColumns to new cookie and remove the old cookie
+                if(cookieConsent) {
+                    var cookieDate = new Date();
+                    cookieDate.setFullYear(cookieDate.getFullYear() + 20);
+                    var savedColumns = $cookies.get(oldName);
+                    $cookies.put(`sc-${newName}`, savedColumns, {path: "/", samesite: "lax", secure: true, expires: cookieDate});
+                    $cookies.remove(`sc-${oldName}`, {"path": "/"});
```

Comment:
If the name doesn't change, this will cause the cookie to be removed completely. Should probably only do this whole cookie update if `oldName !== newName`